proposals/simd-0170-builtin-instruction-cost-and-budget.md

```diff
@@ -0,0 +1,126 @@
+---
+simd: '0170'
+title: Specifying CU Definitions for Builtins
+authors:
+  - Tao Zhu (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2024-08-26
+feature: https://github.com/anza-xyz/agave/issues/2562
+supersedes: 
+superseded-by:
+extends:
+---
+
+## Summary
+
+1. Builtin programs should consume a predefined number of CUs for each
+   instruction.
+2. Since builtins can invoke other programs (CPI), they should allocate enough
+   but granular CUs to ensure successful execution without over-allocation.
```

Comment:
Let's simplify this to something like "If a transaction doesn't define a compute budget limit, then for each builtin program instruction allocate 3000 compute units rather than 200,000."